samples/dnn/dasiamrpn_tracker.py

```diff
@@ -0,0 +1,295 @@
+        top_pad = int(max(0., -context_ymin))
+        right_pad = int(max(0., context_xmax - im_sz[1] + 1))
+        bottom_pad = int(max(0., context_ymax - im_sz[0] + 1))
+
+        context_xmin = context_xmin + left_pad
+        context_xmax = context_xmax + left_pad
+        context_ymin = context_ymin + top_pad
+        context_ymax = context_ymax + top_pad
+        r, c, k = im.shape
+
+        if any([top_pad, bottom_pad, left_pad, right_pad]):
+            te_im = np.zeros((r + top_pad + bottom_pad, c + left_pad + right_pad, k), np.uint8)
+            te_im[top_pad:top_pad + r, left_pad:left_pad + c, :] = im
+            if top_pad:
+                te_im[0:top_pad, left_pad:left_pad + c, :] = self.avg_chans
+            if bottom_pad:
+                te_im[r + top_pad:, left_pad:left_pad + c, :] = self.avg_chans
+            if left_pad:
+                te_im[:, 0:left_pad, :] = self.avg_chans
+            if right_pad:
+                te_im[:, c + left_pad:, :] = self.avg_chans
+            im_patch_original = te_im[int(context_ymin):int(context_ymax + 1), int(context_xmin):int(context_xmax + 1), :]
+        else:
+            im_patch_original = im[int(context_ymin):int(context_ymax + 1), int(context_xmin):int(context_xmax + 1), :]
+
+        if not np.array_equal(model_size, original_sz):
+            im_patch = cv.resize(im_patch_original, (model_size, model_size))
+        else:
+            im_patch = im_patch_original
+
+        return im_patch.astype(float)
+
+def main():
+    parser = argparse.ArgumentParser(description = "Run tracker")
+    parser.add_argument("--net", type = str, help = "Full path to onnx model of net")
+    parser.add_argument("--kernel_r1", type = str, help = "Full path to onnx model of kernel_r1")
+    parser.add_argument("--kernel_cls1", type = str, help = "Full path to onnx model of kernel_cls1")
+    parser.add_argument("--input", type = str, default = " ", help = "Full path to input. Do not use if input is camera")
+    args = parser.parse_args()
+
+    point1 = ()
+    point2 = ()
+    mark = True
+    drawing = False
+    cx, cy, w, h = 0.0, 0.0, 0, 0
+
+    def get_bb(event, x, y, flag, param):
+        nonlocal point1, point2, cx, cy, w, h, drawing, mark
+
+        if event == cv.EVENT_LBUTTONDOWN:
+            if drawing == False:
+                drawing = True
+                point1 = (x, y)
+            else:
+                drawing = False
+        elif event == cv.EVENT_MOUSEMOVE:
+            if drawing == True:
+                point2 = (x, y)
+        elif event == cv.EVENT_LBUTTONUP:
+            cx = point1[0] - (point1[0] - point2[0]) / 2
+            cy = point1[1] - (point1[1] - point2[1]) / 2
+            w = abs(point1[0] - point2[0])
+            h = abs(point1[1] - point2[1])
+            mark = False
+
+    net = cv.dnn.readNet(args.net)
+    kernel_r1 = cv.dnn.readNet(args.kernel_r1)
+    kernel_cls1 = cv.dnn.readNet(args.kernel_cls1)
+
+    if args.input == " ":
```

Comment:
What does it mean? What if input is not empty?

Reply:
"else" is WIP - case for not empty "input".

Reply:
For now it works without drawing rectangle in real time, but correctly draw it in the end and evaluate coefficients correctly .